react-app/src/components/RoomsList.js

```diff
@@ -0,0 +1,37 @@
+import React from 'react'
+import axios from 'axios'
+import Room from './Room'
+
+const getRoomUrl = 'https://hall-booking-system.herokuapp.com/room';
+
+class RoomsList extends React.Component {
+  constructor() {
+    super()
+    this.state = {
+      rooms: [],
+    };
+  }
+
+  componentDidMount() {
+    axios.get(getRoomUrl)
+    .then((response) => {
+      if(response.status === 500) {
+        return
+      }
+      this.setState({
+        rooms: response.data.rooms
+      })
+  });
+  }
+
+  render() {
+    const rooms = this.state.rooms.map(room => <Room key={room._id} roomName={room.roomName} />)
```

Comment:
write a fallback in case axios fails to get `rooms` data
render a "sorry, our servers are facing internal errors. we'll get back to you as soon as possible" or something
together with a picture of a cat

Reply:
There is a check for response status on line 18. Do you want me to put this message there?

Reply:
I think it’ll be efficient to check if this.state.rooms is an empty array